src/components/SlidingNavigation/SlidingNavigation.js

```diff
@@ -0,0 +1,112 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import "./SlidingNavigation.css";
+
+const SlidingNavigation = ({ data, options }) => {
+  const activeItemClassName = `SlidingNavigation__item--active`;
+  const observerOptions = options || {
+    threshold: 0.2,
+  };
+
+  React.useEffect(() => {
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (
+          entry.isIntersecting &&
+          !window.location.hash.includes(entry.target.id)
+        ) {
+          window.history.replaceState({}, ``, `#${entry.target.id}`);
+          const currentActiveElement = document.querySelector(
+            `.${activeItemClassName}`
+          );
+
+          // Remove current active className
+          if (currentActiveElement) {
+            currentActiveElement.classList.remove(activeItemClassName);
```

Comment:
Your'e interacting with real DOM which could cause unexpected behaviour and bugs. If you want to do stuff like this work with node references instead of actual nodes. Virtual DOM vs Real DOM.